ha/core/system_health/system_health.py

```diff
@@ -307,19 +307,85 @@ def publish_event(self, healthevent: HealthEvent, healthvalue: str= ""):
+                else return  False
+
+        """
+        init_health_in_progress = False
+        init_health = self.healthmanager.get_key("init_system_health")
+
+        if (init_health != None) and (init_health == 0):
+            init_health_in_progress = False
+            return init_health_in_progress
+
+        if init_health == None:
+            # Create key and set value to 1
+            self.healthmanager.set_key("init_system_health", "1")
+
+            curr_time = str(int(time.time()))
+            # Currently this timeout is started form the time first event is received till timout
+            # Confirm that this fine.. or should be started from bootstrap time.
+            # ideally both will be almost identical.
+            self.healthmanager.set_key("inital_time", curr_time)
+            timeout = Conf.get(const.HA_GLOBAL_INDEX,  f"SYSTEM_HEALTH{_DELIM}sys_health_bootstrap_timeout")
+            self.healthmanager.set_key("sys_health_bootstrap_timeout", timeout)
+            init_health_in_progress = True
+        else:
+            # check if data for all pods already collected (optional check ;
+            # can be removed if total_num_pods will not be avilable in confstore)
+            if self._check_num_pods():
+                self.healthmanager.set_key("init_system_health", "0")
+                init_health_in_progress = False
+                return init_health_in_progress
+
+            # check if within timeout period
+            curr_time = int(time.time())
+            init_time = int(self.healthmanager.get_key("inital_time"))
+            timeout = int(self.healthmanager.get_key("sys_health_bootstrap_timeout"))
+
+            if curr_time - init_time <= timeout:
+                init_health_in_progress = True
+            else:
+                self.healthmanager.set_key("init_system_health", "0")
+                init_health_in_progress = False
+
+        return init_health_in_progress
+
     def _update(self, healthevent: HealthEvent, healthvalue: str, next_component: str=None):
         """
         update method. This is an internal method for updating the system health.
         """
         component = SystemHealthComponents.get_component(healthevent.resource_type)
         comp_type = healthevent.resource_type.split(':')[-1]
         comp_id = healthevent.resource_id
+
+        bootstrp_in_progress = self._bootstrap_in_progress()
+
         key = self._prepare_key(component, cluster_id=self.node_map['cluster_id'], site_id=self.node_map['site_id'],
                                 rack_id=self.node_map['rack_id'], storageset_id=self.node_map['storageset_id'],
                                 node_id=self.node_id, server_id=self.node_id, storage_id=self.node_id,
                                 comp_type=comp_type, comp_id=comp_id)
         self.healthmanager.set_key(key, healthvalue)
-        self.publish_event(healthevent, healthvalue)
+        if not bootstrp_in_progress:
```

Comment:
If pod xyz online event was received and after some time before initial cluster setup timeout failed event is received for pod xyz, such event needs to be published?

Reply:
This is where the logic in  _check_num_pods would have helped. Without that,  we can not be sure if the failure was during init of the cluster/pod or not; and whether the component expects it to be reported or not. 

We can discuss and conclude on the same

Reply:
This also can be handled in cluster cardinality related changes.